cmd/sidecred/main.go

```diff
@@ -6,40 +6,56 @@ import (
 	"os"
 
 	"github.com/telia-oss/sidecred"
+	"github.com/telia-oss/sidecred/config"
 	"github.com/telia-oss/sidecred/internal/cli"
 
 	"github.com/alecthomas/kingpin"
-	"sigs.k8s.io/yaml"
 )
 
 var version string
 
 func main() {
 	var (
 		app        = kingpin.New("sidecred", "Sideload your credentials.").Version(version).Writer(os.Stdout).DefaultEnvars()
-		namespace  = app.Flag("namespace", "Namespace to use when processing the requests.").Required().String()
 		configPath = app.Flag("config", "Path to the config file containing the requests").ExistingFile()
 		statePath  = app.Flag("state", "Path to use for storing state in a file backend").Default("state.json").String()
 	)
-	cli.Setup(app, runFunc(namespace, configPath, statePath), nil, nil)
+	cli.AddRunCommand(app, runFunc(configPath, statePath), nil, nil).Default()
+
+	validate := app.Command("validate", "Validate a sidecred config.")
+	validate.Action(func(_ *kingpin.ParseContext) error {
+		b, err := ioutil.ReadFile(*configPath)
+		if err != nil {
+			return fmt.Errorf("failed to read config: %s", err)
+		}
+		cfg, err := config.Parse(b)
+		if err != nil {
+			return fmt.Errorf("failed to parse config: %s", err)
```

Comment:
Why not use `kingpin.Fatalf` here as below?

Reply:
We have a few ways to exit with an error:
1. `return fmt.Errorf(...)` inside a `kingpin.Action`. This gets printed as `sidecred: error: <error message>, try --help`.
2. `kingpin.Fatalf(...)` gets printed as `sidecred: error: <error message>`
3. `zap.Fatal(...)` which gets printed as JSON.

I've pushed commits that consolidates this a bit, so that we use `kingpin.Fatalf` for errors in `sidecred validate` since we want these errors to be human readable. And then for the `sidecred run` command I've fixed so that `runFunc` errors uses the logger which prints JSON.

Reply:
Great, makes perfect sense 👍